### PR TITLE
wip

### PR DESCRIPTION
**Pull request by Randgalt**
Moved here: https://github.com/starburstdata/starburst-universe/pull/882

### CODE REVIEW

**Review by amonat — COMMENTED** [review]
I've read the Introduction and Design page so far. I like the writing style and the use of a project to tie the features together.

**Review by amonat — COMMENTED** [review]
A few comments from reading the section on serialization

File: 00-bleeding-edge-java/04-parsing.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+There are a number of wrinkles we'll have to deal with when interpreting characters:
+
+- We sometimes need to look ahead before we know what token type is implied (for literals, strings, numbers)
+- Our library introduces an object name token. This means that when we parse a string we need to wait for the next token. If it's a `:` we have to convert the string into an object name.
+- Number tokens can possibly end with a parseable character: `,` `[` `{`. When this happens we need to reprocess that character.
+
+Here is the general flow for parsing JSON text:
+
+![](assets/parse-flow.png)
```
Comment (petroav) [inline 00-bleeding-edge-java/04-parsing.md:32]: There are two blank white squares at the top left and bottom right corners of the image whose purpose is unclear.

**Review by petroav — COMMENTED** [review]
I've reviewed the first 4 parts and will finish the rest later. I think this blog post is a great idea! I've run all the jshell statements and they work as expected.

File: 00-bleeding-edge-java/01-design.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+- All the record tokens implement `JsonToken`
+- Users of this library cannot define new JsonToken types
+- We can now write code that relies on the fact that there is a known, limited set of JSON token types
+- Notice how little code is needed to fully specify the model. In the past Java was accused of being verbose. It's no longer the case.
+
+With this hierarchy we have an abstract way to describe a JSON object. We can parse JSON text into a list of these
+tokens, or we can destructure a Java object into a list of tokens as shown in this figure:
+
+![](assets/flow.png)
```
[the remark this answers is not in the document]
Comment (Randgalt) [reply 00-bleeding-edge-java/01-design.md:102]: fixed

File: 00-bleeding-edge-java/02-serialization.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        };
+    }
+}
+```
+
+We still need to define `serializeCollection()` and `serializeRecord()` but let's first describe what's going on with the switch statement. 
+
+### Enhanced `switch` and patterns
+
+We're all familiar with `switch` statements from Java and most other languages. In Java 19, switch is much more powerful with these additional features:
```
[the remark this answers is not in the document]
Comment (Randgalt) [reply 00-bleeding-edge-java/02-serialization.md:73]: This was on purpose. Most of the time switch is used as an English word.

File: 00-bleeding-edge-java/04-parsing.md (hunk 1 above)
Comment (Randgalt) [reply 00-bleeding-edge-java/04-parsing.md:32]: Fixed

**Comment by Randgalt** [comment]
This has moved to here: https://github.com/starburstdata/starburst-universe/pull/882

**Pull request closed without merging**